src/init.rs

```diff
@@ -24,30 +33,254 @@ description = ""
+            let default_module_name = Path::new(&module.source)
+                .file_stem()
+                .unwrap()
+                .to_str()
+                .unwrap()
+                .to_owned();
+            module.name = ask_until_valid(
+                " - Name",
+                Some(default_module_name.clone()),
+                util::validate_name,
+            )?;
+            let (abi, interfaces): (Abi, Option<HashMap<String, String>>) = match Select::new()
+                .with_prompt(" - ABI")
+                .item("None")
+                .item("WASI")
+                .item("Emscripten")
+                .default(0)
+                .interact()?
+            {
+                1 => (
+                    Abi::Wasi,
+                    Some(
+                        [("wasi".to_owned(), WASI_LAST_VERSION.to_owned())]
+                            .iter()
+                            .cloned()
+                            .collect(),
+                    ),
+                ),
+                2 => (Abi::Emscripten, None),
+                0 | _ => (Abi::None, None),
+            };
+            module.abi = abi;
+            module.interfaces = interfaces;
+            // We ask for commands if it has an Abi
+            if module.abi == Abi::Wasi || module.abi == Abi::Emscripten {
+                let commands = ask_until_valid(
+                    " - Commmands (space separated)",
+                    Some(default_module_name.clone()),
+                    validate_commands,
+                )?;
+                if !commands.is_empty() {
+                    all_commands.push(Command {
+                        name: commands,
+                        module: module.name.clone(),
+                        main_args: None,
+                        package: None,
+                    });
+                }
+            }
+            all_modules.push(module);
+        }
+        manifest.module = if all_modules.is_empty() {
+            None
+        } else {
+            Some(all_modules)
+        };
+        manifest.command = if all_commands.is_empty() {
+            None
+        } else {
+            Some(all_commands)
+        };
     }
 
-    let mut f = fs::File::create(manifest)?;
-    f.write_all(
-        construct_template_manifest_from_data(util::get_username()?, package_name).as_bytes(),
-    )?;
+    let print_text = if force_yes {
+        "Wrote to"
+    } else {
+        "About to write to"
+    };
+    println!(
+        "\n{} {}:\n\n{}\n",
+        print_text,
+        manifest.base_directory_path.to_str().unwrap(),
+        manifest.to_string()?
+    );
 
+    if force_yes
+        || Confirmation::new()
+            .with_text("Is this OK? (yes)")
+            .default(true)
+            .interact()?
+    {
+        manifest.save()?;
+        #[allow(unused_must_use)]
+        {
+            init_gitignore(manifest.base_directory_path);
```

Comment:
Minor comment: probably worth logging the error unless we really don't care at all